python/sbp/client/loggers/device_iterator.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Mark Fine <mark@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+from base_logger import LogIterator
+from ...client.handler import Handler
+from ...table import dispatch
+import calendar
+import time
+from Queue import Queue, Empty
+
+
+
+class SbpQueue(object):
+  """
+  SbpQueue
+
+  The :class: `SbpQueue` provides a wrapper around Pythons Queue
+
+  Parameters
+  ----------
+  handler : Handler
+    handler object for SBP messages
+  maxsize : int
+    maxsize of the Queue object (0 means infinitie)
+
+  """
+  def __init__(self, handler, maxsize=0 ):
+    self.myqueue = Queue(maxsize)
+
+  def callback(self, msg):
+    """
+    called with each receipt of the message
+    Note: dispatch is necessary but perhaps should be handled in handler
```

Comment:
Interesting - this, JSONLogger, and PickleLogger are all calling `dispatch(msg)`. It might make more sense to move dispatch further up and do it once: https://github.com/swift-nav/libsbp/blob/master/python/sbp/client/handler.py#L83. @mookerji can you think of any reason that would be a bad idea?